src/systems/physics/Physics.cc

```diff
@@ -1325,6 +1347,58 @@ void PhysicsPrivate::UpdatePhysics(EntityComponentManager &_ecm)
         return true;
       });
 
+  // Update model angular velocity
+  _ecm.Each<components::Model, components::AngularVelocityCmd>(
+      [&](const Entity &_entity, const components::Model *,
+          const components::AngularVelocityCmd *_angularVelocityCmd)
+      {
+        auto modelIt = this->entityModelMap.find(_entity);
+        if (modelIt == this->entityModelMap.end())
+          return true;
+
+        auto freeGroup = modelIt->second->FindFreeGroup();
+        if (!freeGroup)
+          return true;
+
+        auto worldAngularVelFeature = entityCast(_entity, freeGroup,
+              this->entityWorldVelocityCommandMap);
+        if (!worldAngularVelFeature)
+        {
+          return true;
+        }
+
+        worldAngularVelFeature->SetWorldAngularVelocity(
+          math::eigen3::convert(_angularVelocityCmd->Data()));
+
+        return true;
+      });
+
+  // Update model linear velocity
+  _ecm.Each<components::Model, components::LinearVelocityCmd>(
+      [&](const Entity &_entity, const components::Model *,
+          const components::LinearVelocityCmd *_linearVelocityCmd)
+      {
+        auto modelIt = this->entityModelMap.find(_entity);
+        if (modelIt == this->entityModelMap.end())
+          return true;
+
+        auto freeGroup = modelIt->second->FindFreeGroup();
+        if (!freeGroup)
+          return true;
+
+        auto worldLinearVelFeature = entityCast(_entity, freeGroup,
+              this->entityWorldVelocityCommandMap);
+        if (!worldLinearVelFeature)
+        {
+          return true;
+        }
+
+        worldLinearVelFeature->SetWorldLinearVelocity(
+          math::eigen3::convert(_linearVelocityCmd->Data()));
```

Comment:
I think you need to multiple `_linearVelocityCmd` by the model's orientation to get the vel in the correct frame. 

May have to do the same for `_angularVelocityCmd`

Reply:
I tried to use `canonicalPoseComp->Data().Pos()` and `canonicalPoseComp->Data().Rot()` to get the orientation and multiply with velocity but it didn't come out right. Is there any example of how to integrate orientation and velocity?

Reply:
this should work:

```
        const components::Pose *poseComp =
            _ecm.Component<components::Pose>(_entity);
        math::Vector3d worldLinearVel = poseComp->Data().Rot() *
            _linearVelocityCmd->Data();

        worldLinearVelFeature->SetWorldLinearVelocity(
          math::eigen3::convert(worldLinearVel));
```

Reply:
Thanks for the suggestion! It worked well. I've modified the velocity to incorporate orientation in https://github.com/ignitionrobotics/ign-gazebo/pull/169/commits/b47e2a50c7ed15ec9078ea7956bc496b0ed84c88 , and am now working on adding tests to boost code coverage

Reply:
great, can you apply that to `AngularVelocityCmd` as well? thanks.